plugins/analytics-module-ga/src/apis/implementations/AnalyticsApi/GoogleAnalytics.test.ts

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ConfigReader } from '@backstage/config';
+import ReactGA from 'react-ga';
+import { GoogleAnalytics } from './GoogleAnalytics';
+
+describe('GoogleAnalytics', () => {
+  const trackingId = 'UA-000000-0';
+  const basicValidConfig = new ConfigReader({
+    app: { analytics: { ga: { trackingId, testMode: true } } },
+  });
+
+  beforeEach(() => {
+    ReactGA.testModeAPI.resetCalls();
+  });
+
+  describe('fromConfig', () => {
+    it('throws when missing trackingId', () => {
+      const config = new ConfigReader({ app: { analytics: { ga: {} } } });
+      expect(() => GoogleAnalytics.fromConfig(config)).toThrowError(
+        /Missing required config value/,
+      );
+    });
+
+    it('returns implementation', () => {
+      const api = GoogleAnalytics.fromConfig(basicValidConfig);
+      expect(api.captureEvent).toBeDefined();
+
+      // Initializes GA with tracking ID.
+      expect(ReactGA.testModeAPI.calls[0]).toEqual([
```

Comment:
This `ReactGA.testModeAPI.calls` is cool maybe we can do something similar with the `MockAnalyticsApi`

Reply:
Wanted to keep `MockAnalyticsApi` similar to other mocked APIs in `test-utils`.  Though I think there are probably some improvements (similar to this) we can make to all of the mocked APIs in there to improve the test DX for all things in Backstage.